client/components/checklist/task.js

```diff
@@ -173,4 +207,7 @@ class Task extends PureComponent {
 	}
 }
 
-export default localize( Task );
+export default connect(
```

Comment:
I have reservations about this. It breaks with [`components`](https://github.com/Automattic/wp-calypso/blob/master/client/components/README.md) vs [`blocks`](https://github.com/Automattic/wp-calypso/blob/master/client/blocks/README.md) guidelines.

> Blocks are React components created from UI primitives to generate more complex entities.
> 
> They are generally connected to state, and have the ability to dispatch actions.

Components are generally trivial to pick up and use because you know (in theory) that you are in control of the component. Blocks are generally trickier because they have interaction with the application beyond your control.

Some questions to try to move this forward:

- Are these analytics critically important?
- Can collapsing behavior be handled by a consumer? (`onExpand` / `onCollapse`?)
- Can we move the collapse analytics handling to a consumer?

Reply:
> Can we move the collapse analytics handling to a consumer?

@sirreal Thanks for the explanation, and the heads up about the guidelines. **:TIL:**

No worries, we can pass down a tracking function as a prop...